llvm/lib/SYCLLowerIR/SYCLVirtualFunctionsAnalysis.cpp

```diff
@@ -71,37 +71,23 @@ void checkKernel(const Function *F, const CallGraphTy &CG) {
 void computeFunctionToKernelsMappingImpl(Function *Kernel, const Function *F,
                                          const CallGraphTy &CG,
                                          FuncToFuncMapTy &Mapping) {
+  Mapping[F].insert(Kernel);
   CallGraphTy::const_iterator It = CG.find(F);
   // It could be that the function itself is a leaf and doesn't call anything
   if (It == CG.end())
     return;
 
-  Mapping[F].insert(Kernel);
```

Comment:
So, when we entered a leaf, we wouldn't have anything in call graph for it and therefore wouldn't update the mapping - is my understanding of the problem correct?

Reply:
Yea, that's correct